src/modules/messaging/index.js

```diff
@@ -137,6 +137,10 @@ export default class Messaging extends ModuleBase {
     return getNativeModule(this).hasPermission();
   }
 
+  registerForRemoteNotifications(): Promise<void> {
```

Comment:
Could we namespace this under ios? It's good to keep platform-specific methods outside of the main cross-platform methods.

e.g. this becomes `firebase.messaging().ios.registerForRemoteNotifications()`